indigo_app/views/works.py

```diff
@@ -82,6 +82,28 @@ def get_work_timeline(self, work):
         # add expressions
         for entry in timeline:
             entry.expressions = [e for e in work_expressions if e.expression_date == entry.date]
+        # add tasks
+        timeline_tasks = work.tasks.filter(timeline_date__isnull=False)
+        dates = [entry.date for entry in timeline]
+        # simple case: add tasks to existing corresponding entries
+        for entry in timeline:
+            entry.tasks = timeline_tasks.filter(timeline_date=entry.date)
```

Comment:
bear in mind that when the template iterates over `entry.tasks`, it will go to the db for each point in time. for something like the income tax act, that could be 50 times.

going to the db is expensive;  we know that we're going to use every task we fetch from the db, so it's better to load all the timeline-related tasks on line 86 and store them in a list, and then split them up in this loop.

Reply:
I'm not sure if I've alleviated your concern here; when you say I should store them in a list and split them up, do you mean by some method other than filtering? If I turn timeline_tasks from a queryset into a list I guess I could iterate over the list and compare each one's date to the entry's? is that more what you were thinking?